Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -435,6 +435,36 @@ class StoreKit1IntegrationTests: BaseStoreKitIntegrationTests {
         try await subscribe()
     }
 
+    func testSubscribeAfterExpirationWhileAppIsClosed() async throws {
+        // 1. Subscribe
+        let customerInfo = try await self.purchaseMonthlyOffering().customerInfo
+        let entitlement = try XCTUnwrap(customerInfo.entitlements[Self.entitlementIdentifier])
+
+        // 2. Simulate closing app
+        Purchases.clearSingleton()
+
+        // 3. Force several renewals while app is closed.
+        for _ in 0..<3 {
+            try self.testSession.forceRenewalOfSubscription(productIdentifier: entitlement.productIdentifier)
+        }
+
+        // 4. Expire subscription
+        try await self.expireSubscription(entitlement)
+
+        // 5. Re-open app
+        await self.resetSingleton()
+        self.logger.clearMessages()
+
+        // 6. Purchase again
+        try await self.purchaseMonthlyProduct()
+
+        self.logger.clearMessages()
+
+        // 7. Verify no transaction is posted as a purchase.
+        self.logger.verifyMessageWasNotLogged("Posting receipt (source: 'purchase')",
+                                              allowNoMessages: true)
```

Comment:
shouldn't we eventually post the transaction from 6 as a purchase?

Reply:
LOL yeah this is a bad refactor from me changing the test `clearMessages` shouldn't be there.